internal/api/cluster.go

```diff
@@ -576,3 +582,86 @@ func handleGetAllUtilityMetadata(c *Context, w http.ResponseWriter, r *http.Requ
 	w.WriteHeader(http.StatusOK)
 	outputJSON(c, w, cluster.UtilityMetadata)
 }
+
+// handleAddClusterAnnotations responds to POST /api/cluster/{cluster}/annotations,
+// adds the set of annotations to the Cluster.
+func handleAddClusterAnnotations(c *Context, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	clusterID := vars["cluster"]
+	c.Logger = c.Logger.WithField("cluster", clusterID).WithField("action", "add-cluster-annotations")
+
+	clusterDTO, status, unlockOnce := lockCluster(c, clusterID)
+	if status != 0 {
+		w.WriteHeader(status)
+		return
+	}
+	defer unlockOnce()
+
```

Comment:
We need to respect the API security lock like we do with other API endpoints.

This should be done for the other endpoints as well.

Reply:
Yes, that makes sense, I will add that.